src/plugins/data/public/ui/filter_bar/filter_item.tsx

```diff
@@ -71,19 +71,27 @@ export function FilterItem(props: Props) {
 
   useEffect(() => {
     const index = props.filter.meta.index;
+    let isSubscribed = true;
```

Comment:
yes, no more warning in the console! however we need to take care of rerendering of this component in a follow up PR, which is I think the reason of  the former warning.

Reply:
I think the reason for the warning was that `getIndexPatterns` is an async task, and once it's done it tries to update the state of the component, but the component may have been unmounted already.  The cleanup will run once the component is unmounted, cleanup will unset the flag, which will prevent subsequent updates to the state. This code is not in any way tampering with the lifecycle of the component. The only question here is if there's a more sophisticated way to get this, beyond plain flag setting, but I'll defer this decision to component owners.